src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -1,116 +1,528 @@
-import React, { useState, useEffect, useCallback } from 'react';
-import { useHistory, useParams } from 'react-router-dom';
-import { Avatar, Grid, TextField, InputLabel } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import {
+  Avatar,
+  Grid,
+  TextField,
+  InputLabel,
+  Box,
+  IconButton,
+  Typography,
+} from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import { RootStateType } from 'models/rootReducer';
+import { BasicButton } from 'components/Form';
+import { PhotoCamera } from '@material-ui/icons';
+import { Autocomplete } from '@material-ui/lab';
+import { uploadImageToImgur } from 'old/lib/utilities/Imgur/uploadImageToImgur';
+import { getStringFromFile } from 'old/lib/utilities/Imgur/getStringFromFile';
+import {
+  getCitiesByRegionId,
+  getRegionByCityId,
+} from 'old/lib/utilities/API/api';
 import { useStyles } from './styles/PersonalInfo.styles';
-import { IExpert } from '../../../old/lib/types';
-import { getCurrentUser } from '../../../old/lib/utilities/API/api';
-import { ERROR_404 } from '../../../old/lib/constants/routes';
+import { ICity, IRegion } from '../../../old/lib/types';
+import { ErrorField } from './ErrorField';
+import { regex } from './constants/regex';
+import { fields } from './constants/fields';
+import { INewAuthorValues } from './types';
 
 export const PersonalInfo: React.FC = () => {
```

Comment:
I think this component has grown too big - not just in lines count but in responsibility. It renders fields, handles changes, does validation, shows errors and loads regions. Too much for single component. Ideally it should handle input and keep local state but delegate validation and should not load regions at all.

Propose to reshape it with extracting region selector into separate component and moving validation logic in separate config object(probably keeping it in the same file, but just out of the component)

what do you say?